### PR TITLE
docs: Review Developer docs

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR reviews the development docs related to this new mono repo structure. I realized that some of those docs were outdated. These docs will be revisited in future PRs.

**Type of change**

(Remember to title the PR according to the type of change)

- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes.)

- [ ] `sphinx-autobuild` (read [Developer Documentation](https://docs.argilla.io/en/latest/community/developer_docs.html#building-the-documentation) for more details)

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: docs/_source/community/developer_docs.md [hunk 1]
```diff
@@ -271,108 +281,64 @@ rm ~/.argilla/argilla.db
...
-
-#### Build Frontend Static Files
-
-Build the static UI files in case you want to work on the UI:
+## Make Your Contribution
 
-```sh
-bash scripts/build_frontend.sh
-```
+Now that everything is up and running, you can start to develop and contribute to Argilla! You can refer to our [contributer guide](/community/contributing.md) to have an understanding of how you can structure your contribution and upload it to the repository.
```
Comment (nataliaElv) [inline docs/_source/community/developer_docs.md]:
```suggestion
Now that everything is up and running, you can start to develop and contribute to Argilla! You can refer to our [contributer guide](/community/contributing.md) to have an understanding of how you can structure your contribution and upload it to the repository.
```
```suggestion
Now that everything is up and running, you can start to develop and contribute to Argilla! You can refer to our [contributor guide](/community/contributing.md) to have an understanding of how you can structure your contribution and upload it to the repository.
```

File: docs/_source/community/developer_docs.md [hunk 2]
```diff
@@ -14,6 +14,13 @@ Being a developer in Argilla means that you are a part of the Argilla community
 
 - **Vue.js UI**: A web application to visualize and annotate your data, users, and teams. It is built with `Vue.js` and is directly deployed alongside the Argilla Server within our Argilla Docker image.
 
+The Argilla repository has a mono repo structure, which means that all the components are in the same repository, and is divided into the following folders:
```
Comment (nataliaElv) [inline docs/_source/community/developer_docs.md]:
```suggestion
The Argilla repository has a monorepo structure, which means that all the components live in the same repository: `argilla-io/argilla`. This repo is divided into the following folders:
```

File: docs/_source/community/developer_docs.md [hunk 3]
```diff
@@ -158,18 +165,21 @@ source .env/bin/activate
 Then, you just need to install Argilla with the command below. Note that we will install it in editable mode using the -e/--editable flag in the `pip` command to avoid having to re-install it on every code modification, but if you’re not planning to modify the code, you can just omit the -e/--editable flag.
 
 ```sh
+cd argilla
```
Comment (nataliaElv) [inline docs/_source/community/developer_docs.md]: Do we really need to add `cd argilla` in this and upcoming blocks? The first block after "Install with `pip`" already covers this step.

File: docs/_source/community/developer_docs.md [hunk 4]
```diff
@@ -271,108 +281,64 @@ rm ~/.argilla/argilla.db
 
 After deleting the database, you will need to run the [database migration](#run-database-migration) task. By following these steps, you’ll have a fresh and clean database to work with.
 
-### Set up the Frontend
+### Set up Argilla Server
 
-If you want to work on the frontend of Argilla, you can do so by following the steps below.
+If you want to work on the server of Argilla, please visit the `argilla-server` [README.md](./argilla-server/README.md) file to see how to set up the server and run it on your local machine.
```
Comment (nataliaElv) [inline docs/_source/community/developer_docs.md]: Seems like the link isn't working.

File: docs/_source/community/developer_docs.md [hunk 5]
```diff
@@ -271,108 +281,64 @@ rm ~/.argilla/argilla.db
...
 
-#### Clone the Argilla Repository
+### Set up Argilla Frontend
 
-Firstly, you have to [fork our repository and clone the fork](<(/community/contributing.md#work-with-a-fork)>) to your computer.
+If you want to work on the frontend of Argilla, please visit the `argilla-frontend` [README.md](./argilla-frontend/README.md) file to see how to set up the frontend and run it on your local machine.
```
Comment (nataliaElv) [inline docs/_source/community/developer_docs.md]: This link isn't working either.

**Review by nataliaElv — COMMENTED** [review]
Leaving some comments here :)

**Review by sdiazlor — COMMENTED** [review]
Just to add that these references don't work:
- [Set up the Frontend](#set-up-the-frontend),
- [Set up the Server](#set-up-the-server),

File: argilla-frontend/README.md [hunk 6]
```diff
@@ -65,6 +67,12 @@ There are different options to get started:
 
 ## 🖥️ FRONTEND
 
+-Before running Argilla frontend server, we need to install Node version 18:
```
Comment (frascuchon) [inline argilla-frontend/README.md]:
```suggestion
-Before running Argilla frontend server, you need to install Node version 18:
```

File: argilla-frontend/README.md [hunk 7]
```diff
@@ -65,6 +67,12 @@ There are different options to get started:
 
 ## 🖥️ FRONTEND
 
+-Before running Argilla frontend server, you need to install Node version 18:
```
Comment (frascuchon) [inline argilla-frontend/README.md]:
```suggestion
- Before running Argilla frontend server, you need to install Node version 18:
```

**Comment by frascuchon** [comment]
@nataliaElv @sdiazlor the changes are ready to review.

**Review by sdiazlor — APPROVED** [review]

File: argilla-frontend/README.md [hunk 8]
```diff
@@ -65,6 +67,12 @@ There are different options to get started:
 
 ## 🖥️ FRONTEND
 
+- Before running Argilla frontend server, you need to install Node version 18:
+
+```bash
+-brew install node@18
```
Comment (frascuchon) [inline argilla-frontend/README.md]:
```suggestion
brew install node@18
```

**Pull request merged**